File: Esquenta/Entities/ItemVenda.cs
using Esquenta.Entities.Interfaces;
using System;

namespace Esquenta.Entities
{
    public class ItemVenda : IBaseEntity
    {
        public virtual int Id { get; protected set; }
        public virtual DateTime DataVenda { get; set; }
        public virtual decimal Valor { get; set; }
        public virtual decimal ValorTotal { get; set; }
        public virtual long Quantidade { get; set; }
        public virtual Venda Venda { get; set; }
        public virtual Produto Produto { get; set; }
    }
}

File: V2-Node/esquenta.v2/server/models/itemvenda.js
'use strict'
module.exports = (sequelize, DataTypes) => {
  var itemVenda = sequelize.define('itemVenda', {
    quantidade: DataTypes.INTEGER,
    valor: DataTypes.DECIMAL,
    dataVenda: DataTypes.DATE,
    valorTotal: DataTypes.DECIMAL
  }, {})
  itemVenda.associate = function (models) {
    itemVenda.belongsTo(models.venda, {
      foreignKey: 'vendaId',
      onDelete: 'CASCADE'
    })

    itemVenda.belongsTo(models.produto, {
      foreignKey: 'produtoId',
      as: 'Item'
    })
  }
  return itemVenda
}
